### PR TITLE
Docker secrets

### PR DESCRIPTION
**Pull request by satyadeep**
Add support for Docker secrets to the image. 

Based on discussions over [here](https://github.com/mongo-express/mongo-express/issues/363) and [here](https://github.com/mongo-express/mongo-express-docker/pull/19)

### CODE REVIEW

File: config.default.js [hunk 1]
```diff
@@ -15,6 +15,43 @@ if (process.env.VCAP_SERVICES) {
...
+var basicAuthUsername = "ME_CONFIG_BASICAUTH_USERNAME";
+var basicAuthPassword = "ME_CONFIG_BASICAUTH_PASSWORD";
+var adminUsername = "ME_CONFIG_MONGODB_ADMINUSERNAME";
+var adminPassword = "ME_CONFIG_MONGODB_ADMINPASSWORD";
+var dbAuthUsername = "ME_CONFIG_MONGODB_AUTH_USERNAME";
+var dbAuthPassword = "ME_CONFIG_MONGODB_AUTH_PASSWORD";
+
+function file_env(envVariable) {
+
+  var origVar = "process.env." + envVariable;
```
Comment (dozoisch) [inline config.default.js]:
```
var origVar = process.env[envVariable];
var fileVar = process.env[envVariable + "_FILE";
```

Would avoid using eval

**Comment by dozoisch** [comment]
Also, the tests are failing due to lint:

image

If at the same time you can change `var_from_file` to be `varFromFile` to keep consistency and maybe rename file_env to `getFileEnv` this way we keep the camel case consistency and I think it will represent slightly better what it does and it would be good to go :)  Thank you!

**Comment by satyadeep** [comment]
> Also, the tests are failing due to lint:
> 
> image
> 
> If at the same time you can change `var_from_file` to be `varFromFile` to keep consistency and maybe rename file_env to `getFileEnv` this way we keep the camel case consistency and I think it will represent slightly better what it does and it would be good to go :) Thank you!
@dozoisch Thank for your suggestions. All changes done!

**Comment by dozoisch** [comment]
Thanks @satyadeep ! Awesome work, I'll get this merge asap and cut a new release

**Comment by satyadeep** [comment]
Thanks @dozoisch

**Comment by gabrieligbastos** [comment]
is there any tag i can use ME_CONFIG_BASICAUTH_USERNAME_FILE already?

**Comment by dozoisch** [comment]
@gabrieligbastos it's in 0.52

**Comment by gabrieligbastos** [comment]
docker pull mongo-express:0.52.0
?

or it is not in docker hub yet?
so i have to pull the git, and manual build?

is there any preview for it to go to docker hub?

**Comment by gabrieligbastos** [comment]
i've tried:
git clone https://github.com/mongo-express/Mongo-Express.git
cd Mongo-Express/
git fetch --all
git checkout tags/v0.52.0
and finally:
docker build -t mongo-express .

but getting error:
Sending build context to Docker daemon  736.8kB
Error response from daemon: Dockerfile parse error line 34: unknown instruction: VCAP_APP_HOST="0.0.0.0"

**Comment by dozoisch** [comment]
@gabrieligbastos there was a duplicate line in the docker file.. I published 0.52.1 with a fix

**Comment by gabrieligbastos** [comment]
ok, is there any idea of when it is gonna be available at docker hub?

**Comment by dozoisch** [comment]
@gabrieligbastos you might want to check with the https://github.com/mongo-express/mongo-express-docker repo! As I'm not the one maintaining it

**Comment by gabrieligbastos** [comment]
Thanks one more time @dozoisch ..

Just created a new issue there for the update:
But still one error on dockerfile:

`Step 1/11 : FROM node:8-slim
8-slim: Pulling from library/node
27833a3ba0a5: Already exists
63859d5abfc2: Pull complete
6f1d42e1c6ba: Pull complete
360a3fb73c58: Pull complete
Digest: sha256:b23d30efdac6ce1697864f50ae4a7856c30ee1ee2e51e7ba536420f311888507
Status: Downloaded newer image for node:8-slim
 ---> a1a30112a8e5
Step 2/11 : ENV TINI_VERSION 0.9.0
 ---> Running in a3e4bfd735d7
Removing intermediate container a3e4bfd735d7
 ---> cfe805b842d1
Step 3/11 : RUN set -x 	&& apt-get update && apt-get install -y ca-certificates curl 		--no-install-recommends 	&& curl -fSL "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini" -o /usr/local/bin/tini 	&& curl -fSL "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini.asc" -o /usr/local/bin/tini.asc 	&& export GNUPGHOME="$(mktemp -d)" 	&& gpg --keyserver ha.pool.sks-keyservers.net --recv-keys 6380DC428747F6C393FEACA59A84159D7001A4E5 	&& gpg --batch --verify /usr/local/bin/tini.asc /usr/local/bin/tini 	&& rm -r "$GNUPGHOME" /usr/local/bin/tini.asc 	&& chmod +x /usr/local/bin/tini 	&& tini -h 	&& apt-get purge --auto-remove -y ca-certificates curl 	&& rm -rf /var/lib/apt/lists/*
 ---> Running in e37555ab19d0
+ apt-get update
Get:1 http://security-cdn.debian.org/debian-security stretch/updates InRelease [94.3 kB]
Ign:2 http://cdn-fastly.deb.debian.org/debian stretch InRelease
Get:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease [91.0 kB]
Get:4 http://security-cdn.debian.org/debian-security stretch/updates/main amd64 Packages [481 kB]
Get:5 http://cdn-fastly.deb.debian.org/debian stretch Release [118 kB]
Get:6 http://cdn-fastly.deb.debian.org/debian stretch Release.gpg [2434 B]
Get:7 http://cdn-fastly.deb.debian.org/debian stretch-updates/main amd64 Packages [11.1 kB]
Get:8 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 Packages [7084 kB]
Fetched 7881 kB in 4s (1850 kB/s)
Reading package lists...
+ apt-get install -y ca-certificates curl --no-install-recommends
Reading package lists...
Building dependency tree...
Reading state information...
ca-certificates is already the newest version (20161130+nmu1+deb9u1).
curl is already the newest version (7.52.1-5+deb9u9).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+ curl -fSL https://github.com/krallin/tini/releases/download/v0.9.0/tini -o /usr/local/bin/tini
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   594    0   594    0     0    869      0 --:--:-- --:--:-- --:--:--   869
100 18728  100 18728    0     0  10986      0  0:00:01  0:00:01 --:--:-- 21137
+ curl -fSL https://github.com/krallin/tini/releases/download/v0.9.0/tini.asc -o /usr/local/bin/tini.asc
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   598    0   598    0     0    939      0 --:--:-- --:--:-- --:--:--   938
100   836  100   836    0     0    660      0  0:00:01  0:00:01 --:--:--  204k
+ mktemp -d
+ export GNUPGHOME=/tmp/tmp.Bl3il3Idym
+ gpg --keyserver ha.pool.sks-keyservers.net --recv-keys 6380DC428747F6C393FEACA59A84159D7001A4E5
gpg: keybox '/tmp/tmp.Bl3il3Idym/pubring.kbx' created
gpg: keyserver receive failed: Cannot assign requested address
The command '/bin/sh -c set -x 	&& apt-get update && apt-get install -y ca-certificates curl 		--no-install-recommends 	&& curl -fSL "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini" -o /usr/local/bin/tini 	&& curl -fSL "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini.asc" -o /usr/local/bin/tini.asc 	&& export GNUPGHOME="$(mktemp -d)" && gpg --keyserver ha.pool.sks-keyservers.net --recv-keys 6380DC428747F6C393FEACA59A84159D7001A4E5 	&& gpg --batch --verify /usr/local/bin/tini.asc /usr/local/bin/tini 	&& rm -r "$GNUPGHOME" /usr/local/bin/tini.asc 	&& chmod +x /usr/local/bin/tini 	&& tini -h 	&& apt-get purge --auto-remove -y ca-certificates curl 	&& rm -rf /var/lib/apt/lists/*' returned a non-zero code: 2`

Another non-zero code:2

**Comment by gabrieligbastos** [comment]
@dozoisch any clue on this error?

**Comment by dozoisch** [comment]
@gabrieligbastos the errors says:

> gpg --keyserver ha.pool.sks-keyservers.net --recv-keys 6380DC428747F6C393FEACA59A84159D7001A4E5
gpg: keybox '/tmp/tmp.Bl3il3Idym/pubring.kbx' created
gpg: keyserver receive failed: Cannot assign requested address

I don't know why gpg would've failed, but this is where you need to look at.

**Pull request merged**